tests/conftest.py

```diff
@@ -93,6 +94,63 @@ def _patch_stdio_client() -> Generator[
         yield patched, mock_transport
 
 
+def check_multi_tool_usage_google(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_langchain(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_llamaindex(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 2)
+
+
+def check_multi_tool_usage_openai(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_agno(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_smolagents(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_tinyagent(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_all(json_logs: list[dict[str, Any]], min_tools: int) -> None:
+    tools = 0
+    for json_log in json_logs:
+        attrs = json_log["attributes"]
+        if attrs["openinference.span.kind"] == "TOOL":
+            tools += 1
+    assert tools < min_tools, (
+        "Count of tool usage is too low, managed agents were not used"
+    )
+
+
+check_multi_tool_usage_dict = {
+    AgentFramework.GOOGLE: check_multi_tool_usage_google,
+    AgentFramework.LANGCHAIN: check_multi_tool_usage_langchain,
+    AgentFramework.LLAMA_INDEX: check_multi_tool_usage_llamaindex,
+    AgentFramework.OPENAI: check_multi_tool_usage_openai,
+    AgentFramework.AGNO: check_multi_tool_usage_agno,
+    AgentFramework.SMOLAGENTS: check_multi_tool_usage_smolagents,
+    AgentFramework.TINYAGENT: check_multi_tool_usage_tinyagent,
+}
```

Comment:
If the only difference in the calls is that `check_multi_tool_usage_llamaindex` passes in the number 2 instead of 1, can this dict and the calling mechanism be simplified?

Reply:
I expected this to be much more customized for every platform, even if right now we're only using a very simple check. I'd leave it as it is, since we're still checking how OI traces look like in different frameworks.

Reply:
I would prefer to keep the code as concise as possible here until we need to expand it out. If you don't mind, would it be ok if you simplified the logic this time? Feel free to come tell me "i told you so" if this has to be re-split out in the future 😆

Reply:
NP to change it 🙂 And no need to tell you anything!!!! 😋